tracer/src/Datadog.Trace/Configuration/ImmutableTracerSettings.cs

```diff
@@ -587,6 +589,26 @@ public static ImmutableTracerSettings FromDefaultSources()
             return new ImmutableTracerSettings(TracerSettings.FromDefaultSourcesInternal(), true);
         }
 
+        // Overriding the default "record" behaviour here
+        // This type _shouldn't_ be treated as a record generally, we only made it a record
+        // so we could use with {} expressions...
+
+        /// <inheritdoc />
+        // ReSharper disable once BaseObjectGetHashCodeCallInGetHashCode
+        public override int GetHashCode() => base.GetHashCode();
```

Comment:
Does this one (and `Equals`) need a `[PublicApi]` attribute or is that attribute added _only_ to public members that we want to get telemetry on?